wallet/chain/p/builder.go

```diff
@@ -570,88 +570,90 @@ func (b *builder) spend(
 		Addrs:     []ids.ShortID{addr},
 	})
 
+	// ! @evlekht there is no way to determine if output is locked without
```

Comment:
We need to elobarate this, because we also need this information in our (JS) wallet.
I was in opinion, that we have the bond / deposit state applied on each UTXO (??)

Reply:
> We need to elobarate this, because we also need this information in our (JS) wallet.
We can easily updated GetUTXOs handler so it will request lock state for utxos and append this info to response.
> I was in opinion, that we have the bond / deposit state applied on each UTXO (??)
We didn't change avax.UTXO or secp256k1fx.TransferOutput, instead we keep 3 maps (bi-map like) in chain state: depositTxID -> utxoIDs, bondTxID -> utxoIDs, utxoID -> {depositTxID, bondTxID}

But this wallet thing in caminogo is completely different story - this is another big codebase, I tried to dig into it, but only briefly, and to support this - is another big task. And if we won't use it, I'd like to not do it